ipn/ipnlocal/local.go

```diff
@@ -1815,16 +1817,20 @@ func (b *LocalBackend) Start(opts ipn.Options) error {
 
 	blid := b.backendLogID.String()
 	b.logf("Backend: logs: be:%v fe:%v", blid, opts.FrontendLogID)
-	b.send(ipn.Notify{BackendLogID: &blid})
-	b.send(ipn.Notify{Prefs: &prefs})
+	b.sendLocked(ipn.Notify{BackendLogID: &blid, Prefs: &prefs})
 
-	if !loggedOut && (b.hasNodeKey() || confWantRunning) {
+	shouldStartLogin := false
+	if !loggedOut && (b.hasNodeKeyLocked() || confWantRunning) {
 		// Even if !WantRunning, we should verify our key, if there
 		// is one. If you want tailscaled to be completely idle,
 		// use logout instead.
+		shouldStartLogin = true
+	}
+	b.stateMachineLockedOnEntry(unlock)
+
+	if shouldStartLogin {
```

Comment:
But now we're unlocked and this cc might not be ours? But I guess it'll get canceled?

Reply:
yeah it should get shutdown